### DISCUSSION TITLE
Whats the better way to wait cypress for all XHR requests to complete

### DISCUSSION THREAD

**Discussion by prashant-kajale** [cypress-io/cypress#16162] [General]
While navigating between pages, and also after login to app, there is various XHR request that happens.
SO what the better way to wait till all the requests are complete.
The wait requires to pass the request in it but is impossible to define for each navigation.

**Comment by bahmutov** [comment]
I would suggest waiting for something on the page, like a CSS class being set when all info has finished loading. Unfortunately there is no generic "wait for all network calls to finish" in Cypress.

**Reply by Splines** [reply]
that's unfortunate :(

**Reply by krisodb** [reply]
in the mean time there is following plugin https://github.com/bahmutov/cypress-network-idle

**Reply by Splines** [reply]
However, the way I understand `cypress-network-idle` is that it's adding timeouts which will slow down your tests.

So our workaround is to wait for an HTML element that we expect to show anyways. Doesn't hurt to test that as well. And if you rely on some network requests being finished, stub them using [`cy.intercept()`](https://docs.cypress.io/guides/guides/network-requests).

**Comment by bahmutov** [comment]
You could wait for idle of zero so it would essentially just wait for the network requests to complete without any extra timeSent from my iPhoneOn Sep 18, 2024, at 15:36, Splines ***@***.***> wrote:﻿
However, the way I understand cypress-network-idle is that it's adding timeouts which will slow down your tests.

**Reply by Splines** [reply]
Ah, that's nice. Then I understood this sentence wrong in [your documentation](https://github.com/bahmutov/cypress-network-idle?tab=readme-ov-file#use):

> Wait for two seconds to pass without any network calls (Ajax, static resources)

I thought this meant that you'd have to set the threshold time high enough to be sure you didn't miss any network call that might happen later.

**Comment by bahmutov** [comment]
Well that too - do you want to wait for the current network requests only or any that might be kicked off right after? In any case, feel free to open a new issue in the plugins repository and we can discuss it there Sent from my iPhoneOn Sep 18, 2024, at 15:41, Splines ***@***.***> wrote:﻿
Ah, that's nice. Then I understood this sentence wrong in your documentation:

Wait for two seconds to pass without any network calls (Ajax, static resources)

I thought this meant that you'd have to set the threshold time high enough to be sure you didn't miss any network call that might happen later.

**Reply by Splines** [reply]
To be honest, I'm not too sure anymore in which scenario we needed this in the first place ;) I think we wanted to wait for *any* network request to have finished, but it turned out this wasn't even necessary. Instead, we waited for an HTML element to show and addtionally stubbed one network request for the test at hand. Other requests were not important at that point and they could just as well not be completed at the time Cypress continues. So that's just to tell , I'm actually not facing any problems here anymore 😅
